src/macrotools.jl

```diff
@@ -0,0 +1,90 @@
+using MacroTools
+
+const STRUCTSYMBOL = VERSION < v"0.7-" ? :type : :struct
+
+struct ParseError <: Exception
+    msg::String
+end
+Base.showerror(io::IO, ex::ParseError) = print(io, "ParseError: $(ex.msg)")
+
+function parse_error(ex)
+    throw(ParseError("Cannot parse typedefinition from $ex."))
+end
+
+function splittypedef(ex)
+    ex = MacroTools.striplines(ex)
+    ex = MacroTools.flatten(ex)
+    d = Dict{Symbol, Any}()
+    if @capture(ex, struct header_ body__ end)
+        d[:mutable] = false
+    elseif @capture(ex, mutable struct header_ body__ end)
+        d[:mutable] = true
+    else
+        parse_error(ex)
+    end
+    
+    if @capture header nameparam_ <: super_
+        nothing
+    elseif @capture header nameparam_
+        super = :Any
+    else
+        parse_error(ex)
+    end
+    d[:supertype] = super
+    if @capture nameparam name_{param__}
+        nothing
+    elseif @capture nameparam name_
+        param = []
+    else
+        parse_error(ex)
+    end
+    d[:name] = name
+    d[:params] = param
+    d[:fields] = []
+    d[:constructors] = []
+    for item in body
+        if @capture item field_::T_
+            push!(d[:fields], (field, T))
+        elseif item isa Symbol
+            push!(d[:fields], (item, Any))
+        else
+            push!(d[:constructors], item)
+        end
+    end
+    d
+end
+
+
+function is_trivial_struct(ex)
+    try
+        return isempty(splittypedef(ex)[:constructors])
+    catch err
+        if err isa ParseError
+            return false
+        end
+        throw()
+    end
+end
```

Comment:
My plan was to move splitt/combine typedef file into a PR to `MacroTools`, once we think its good enough. And the `macrotools.jl` file was meant just for things that need to be in this PR. I should have added a comment at the top of the file, sorry. Could you move `is_trivial_struct` elsewhere?

Reply:
I see.  Should I revert macrotools.jl to the previous state?  I also introduced `ParseError` but I think PR for exposing an exception type as a public API has lower chance to get into MacroTools.jl.